packages/cli/lib/services/response-saver.service.ts

```diff
@@ -88,18 +92,20 @@ export function onAxiosRequestFulfilled({
             customFilePath: 'mocks/nango/getConnection.json'
         });
 
-        saveResponse<Metadata>({
-            directoryName,
-            data: connection.metadata as Metadata,
-            customFilePath: 'mocks/nango/getMetadata.json'
-        });
+        if (!hasStubbedMetadata) {
+            saveResponse<Metadata>({
+                directoryName,
+                data: connection.metadata as Metadata,
+                customFilePath: 'mocks/nango/getMetadata.json'
+            });
+        }
 
         return response;
     }
 
     const requestIdentity = computeConfigIdentity(response.config);
 
-    const syncSubDir = syncVariant ? `${syncName}/${syncVariant}` : syncName;
+    const syncSubDir = syncVariant && syncVariant !== BASE_VARIANT ? `${syncName}/${syncVariant}` : syncName;
```

Comment:
does that mean we cannot snapshot the responses in variant sync? Maybe it is not a problem

Reply:
I haven't tested it actually so I'm not sure we can or not. We should support it if it doesn't work but I think we should cross that path later.

Reply:
up to you. maybe add a TODO to mention it'd be a limitation and variant are not supported